### PR TITLE
#176. Update User Profile UI.

### PR DESCRIPTION
**Pull request by karavaiev-gh**

### CODE REVIEW

File: ios/Codeforces Watcher/Application/features/users/UserViewController.swift [hunk 1]
```diff
@@ -116,25 +131,39 @@ class UserViewController: UIViewControllerWithCross {
...
-            $0.topToBottom(of: currentRatingLabel, offset: 2)
+        starIcon.run {
+            $0.width(16)
+            $0.height(16)
+            $0.topToBottom(of: ratingIcon, offset: 1)
             $0.leading(to: rankLabel)
         }
         
+        contributionLabel.run {
+            $0.topToBottom(of: ratingLabel, offset: 4)
```
Comment (yevhenii-kanivets) [inline ios/Codeforces Watcher/Application/features/users/UserViewController.swift:163]: It's better to use `centerY(to: starIcon)` here.

File: ios/Codeforces Watcher/Application/features/users/UserViewController.swift [hunk 2]
```diff
@@ -154,9 +183,9 @@ class UserViewController: UIViewControllerWithCross {
         
         let none = "None".localized
         
-        rankLabel.text = "Rank".localizedFormat(args: user.rank ?? none)
-        nameLabel.text = "Name".localizedFormat(args: user.firstName ?? "", user.lastName ?? none)
-        currentRatingLabel.text = "Current rating".localizedFormat(args: user.rating ?? none)
-        maxRatingLabel.text = "Max rating".localizedFormat(args: user.maxRating ?? none)
+        rankLabel.attributedText = colorTextByUserRank(text: user.rank?.capitalized ?? none, rank: user.rank)
+        handleLabel.attributedText = colorTextByUserRank(text: user.handle, rank: user.rank)
+        ratingLabel.attributedText = ratingViewByUserRank(text: "Rating".localizedFormat(args: user.rating ?? none, user.maxRating ?? none), currentRank: user.rank, maxRank: user.maxRank)
```
Comment (yevhenii-kanivets) [inline ios/Codeforces Watcher/Application/features/users/UserViewController.swift:188]: Way too long: try to split some code to variables first.

File: ios/Codeforces Watcher/Application/features/users/UserViewController.swift [hunk 3]
```diff
@@ -154,9 +183,9 @@ class UserViewController: UIViewControllerWithCross {
...
+        contributionLabel.attributedText = contributionView(text: "Contribution".localizedFormat(args: user.contribution ?? none))
```
Comment (yevhenii-kanivets) [inline ios/Codeforces Watcher/Application/features/users/UserViewController.swift:189]: Can be simplified with

```
contributionLabel.attributedText = user.contributionText

extension User {
    
    var contributionText: String {
        if let contribution = user.contribution {
            return contributionView(text: "Contribution".localizedFormat(args: contribution))
        } else {
            return AttributedString(text: "no_contribution".localized)
        }
    }
}
```

Don't hesitate to write more code if it improves readability.

**Review by yevhenii-kanivets — CHANGES_REQUESTED** [review]
Looks exactly as on design, but code can be improved. See my comments for details and don't hesitate to ping me for any clarifications.

File: ios/Codeforces Watcher/Application/features/users/UserViewController.swift [hunk 4]
```diff
@@ -116,25 +131,39 @@ class UserViewController: UIViewControllerWithCross {
...
+        ratingIcon.run {
+            $0.width(16)
+            $0.height(16)
+            $0.topToBottom(of: handleLabel, offset: 10)
             $0.leading(to: rankLabel)
         }
+  
+        ratingLabel.run {
+            $0.topToBottom(of: handleLabel, offset: 11)
+            $0.leftToRight(of: ratingIcon, offset: 4)
```
Comment (MatyashDen) [inline ios/Codeforces Watcher/Application/features/users/UserViewController.swift:152]: leadingToTrailing?
@yevhenii-kanivets

Comment (yevhenii-kanivets) [reply ios/Codeforces Watcher/Application/features/users/UserViewController.swift:152]: Yes 😉 (didn't want to complicate the stuff at first)

**Comment by yevhenii-kanivets** [comment]
Copied in https://github.com/xorum-io/codeforces_watcher/pull/198. Rebase was really unmanageable for some reason. Good job 👍

**Pull request closed without merging**